### ISSUE TITLE
"Setup GEOGRID" functionality

### ISSUE THREAD

**Issue by crisjf** [CityScope/CS_Brix#34]
I just had a call with @doorleyr about integrating some functions to setup the GEOGRID. 

These functions will allow the user to:
* Set up the initial state of the grid 
* Modify types-definitions using

Let's track the progress in this issue.

**Comment by crisjf** [comment]
Issues #22 and #23 are related to this.

**Comment by guadalupebabio** [comment]
if there is a quick function to check if a table exists, that'd be super useful. I'm working on a script that creates tables

**Comment by RELNO** [comment]
GET http://cityio.media.mit.edu/api/tables/list for all tables

**Comment by guadalupebabio** [comment]
I know that one, but I was wondering if there is a function in Brix that checks all the existing tables and returns True or False. The same that exists `geogrid = H.get_GEOGRID()`

**Comment by guadalupebabio** [comment]
Also, I was checking [Ronan's code](https://github.com/CityScope/CS_Grid_Maker/blob/master/examples/corktown_grid.py) to generate a new table  and uses a `Grid` function, `corktown_grid= Grid(top_left_lon, top_left_lat, rotation,crs_epsg, cell_size, rows, columns)` is it in included or soon to be in Brix? otherwise how can I create a table from a script having `(top_left_lon, top_left_lat, rotation,crs_epsg, cell_size, rows, columns)`?

**Comment by RELNO** [comment]
```
import requests
import json

def find_if_table_exist(table_name):
    x = requests.get('http://cityio.media.mit.edu/api/tables/list')
    
    list_of_table=x.content.decode('utf-8')
    list_to_json = json.loads(list_of_table)
    
    for this_table in list_to_json:
        clean_table_name = this_table.replace('https://cityio.media.mit.edu/api/table/', '')
        if clean_table_name ==table_name:
            return True
    return False
        
print (find_if_table_exist('cityscopejs'))
```

**Comment by crisjf** [comment]
Check `is_table` function: [here](https://cityscope.media.mit.edu/CS_Brix/functions.html)

All these functions are very straightforward to build, but they are a bit time-consuming, especially to get them working reliably. 

I think we're failing at keeping these helper-like functions in one place. For example, I was looking at @guadalupebabio's code to generate a new table, and the line she was using to post the `GEOGRID` object is not going to be compatible with the new version of cityio. She was also importing some functions from @doorleyr 's repo (the `Grid` function) that could easily live in `brix`), which adds complexity and is prone to error.

**Comment by crisjf** [comment]
@doorleyr should we start importing some of the functions and classes from this repo?
https://github.com/CityScope/CS_Grid_Maker

**Comment by crisjf** [comment]
I just pulled the `grid_geojson.py` from https://github.com/CityScope/CS_Grid_Maker and put it in a new branch `grid_maker` for testing. 

You can now do `from brix import Grid` if you are in the right branch.

I also kept the history of the file.

**Comment by RELNO** [comment]
@crisjf I haven't looked into @doorleyr lib for sometime now, but there might minor differences between the way CSjs editor creates grid and how this library does -- I'd suggest creating the same grid in both and testing for differences

**Comment by crisjf** [comment]
Good to know. I know @guadalupebabio was hoping to use this feature, so she'll let us know if/when it crashes. 

@guadalupebabio can you also try what @RELNO recommended while you're testing?

I also imported the `README` from the other repo:
https://github.com/CityScope/CS_Brix/blob/grid_maker/README_Grid_Maker.md

Once this gets pulled to master we'll integrate this very well written documentation in the docs for `brix`.

**Comment by crisjf** [comment]
I integrated @guadalupebabio changes into `grid_maker.py`

Here's the commit (I linked it to the wrong issue by mistake):
50bf194581f81480a6909070514f780e4602f402

One small note, `commit_grid` was giving the user the option to save the json locally. I removed this, as it should be done by the user independently from the commit.

**Comment by crisjf** [comment]
I had to remove `pyproj` from requirements because it was conflicting with the version required by `geopandas`. `pyproj` is still installed because we require `geopandas`, but @doorleyr code was written for `pyproj==1.9.5` and they are in version `3` by now. So things might break.

**Comment by crisjf** [comment]
See #42 

Thanks @doorleyr for looking into this!

**Comment by crisjf** [comment]
Since @doorleyr closed #42, we'll be integrating these features to the master branch soon :)

**Comment by crisjf** [comment]
I started a PR for this feature.

I'll start working on documentation in the meantime. 

I know we don't have the resources to do an in-depth code review, but I'll leave it up for a few days just in case, and then just merge to master.

**Comment by crisjf** [comment]
I just merged PR #47 

Docs are here:
https://cityscope.media.mit.edu/CS_Brix/tutorial.html#creating-a-table-from-python
